src/System.Drawing.Common/src/System/Drawing/Printing/PrintingServices.Unix.cs

```diff
@@ -75,11 +75,15 @@ internal static PrinterSettings.StringCollection InstalledPrinters
             {
                 LoadPrinters();
                 PrinterSettings.StringCollection list = new PrinterSettings.StringCollection(Array.Empty<string>());
-                foreach (object key in installed_printers.Keys)
+                lock (installed_printers.SyncRoot)
```

Comment:
Note that in .NET Core, SyncRoot for the major collections is just `this`
`        public virtual object SyncRoot => this;`
So you can just use `list` here and it's a little simpler.